packages/builder/src/builder.js

```diff
@@ -667,6 +688,11 @@ export default class Builder {
         if (['add', 'change', 'unlink'].includes(event) === false) {
           return
         }
+        /* istanbul ignore if */
+        if (this.serverMiddlewarePaths.includes(fileName)) {
```

Comment:
This getter is being computed after every file change. Maybe we can somehow cache it...

Reply:
Agree, I think `this.serverMiddlewarePaths` can be assigned to variable at beginning of `watchRestart`, then it will be shared by watcher and also re-calculated after every building

Reply:
Good catch, fixed!